### DISCUSSION TITLE
Reg: - using S3 for storage

### DISCUSSION THREAD

**Discussion by RVaidhy** [openobserve/openobserve#1988] [Ideas]
### Which OpenObserve functionalities are relevant/related to the feature request?

ingestion, other

### Description

Hi Team,

We are planning to implement Openobserve as a logging tool for our prod environment. We are using AWS Cloud.

We are planning to go with single EC2 setup and later this will be converted into HA setup. 

Here we have few queries, please have this answered. so that we shall plan things easily. 

1, How to create a cluster in EC2 as a multi tenant (iExample, if a AZ goes down the another instance from the other AZ will receive the load)

2. How to configure HA and store it in postgres so that the data will be with us for other analytics purpose as well. 

Thanks,
Vaidhy R

### Proposed solution

N/A

### Alternatives considered

N/A

**Comment by prabhatsharma** [comment]
We don't have cloudformation/terraform script for now that will allow you to create a cluster on ec2 directly in an easy way. Of course you could take the difficult route and install things manually, but it would be a good amount of work. We would seriously recommend setting up an EKS cluster if you can. Will see if I can create a cloudformation/terraform script. But that may take time as it needs to be tested and vetted for production use even after initial development is done.

You can't store data in postgres directly. You can however store data in s3 and metadata in postgres. Once you have data in s3 you can use aws glue crawler to crawl the data (which will create schema in glue catalog) and then run sql queries on the data using Amazon athena. You could also run spark on the same data for analysis if desired.

For a single instance deployment that can later be converted into an HA cluster, you could use either postgres or dynamodb for storing metadata and use s3 for storing data.

Once you decide to move to HA mode, just give the same details of postgres/s3 or dynamodb/s3 to the new HA cluster and it will work seamlessly.

Take a look at https://openobserve.ai/docs/ha_deployment/#postgresql on how to use postgres or dynamo as metadata store.
